File: rb/features.rb
# Listloco SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module ListlocoFeatures
  def self.make_feature(name)
    case name
    when "base"
      ListlocoBaseFeature.new
    when "ratelimit"
      ListlocoRatelimitFeature.new
    when "retry"
      ListlocoRetryFeature.new
    when "test"
      ListlocoTestFeature.new
    when "timeout"
      ListlocoTimeoutFeature.new
    else
      ListlocoBaseFeature.new
    end
  end
end

File: py/features.py
# Listloco SDK feature factory

from feature.base_feature import ListlocoBaseFeature
from feature.test_feature import ListlocoTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ListlocoBaseFeature(),
        "test": lambda: ListlocoTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
